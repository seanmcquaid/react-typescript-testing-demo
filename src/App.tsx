import React, { useState, useCallback } from 'react';
import './App.css';
import TextInput from './components/TextInput/TextInput';
import styled from 'styled-components';
import Button from './components/Button/Button';


const App: React.FC = () => {
  const [inputText, setInputText] = useState<string>("");
  const [textAdded, setTextAdded] = useState<Array<String>>([]);

  const inputTextOnChange = useCallback((event: React.ChangeEvent<HTMLInputElement>) => {
    setInputText(event.target.value);
  }, []);

  const buttonOnClick = useCallback(() => {
    setTextAdded([...textAdded, inputText]);
    setInputText("");
  },[textAdded, inputText]);


  return (
    <div className="App">
      <H1>Testing Tutorial</H1>
      <TextInput
        value={inputText}
        onChange={inputTextOnChange}
      />
      <Button
        title="Click Me!"
        onClick={buttonOnClick}
      />
      <ul>
        {textAdded.map((item, i) => <li key={i} data-testid={"itemNumber" + i}>{item}</li>)}
      </ul>
    </div>
  );
}

const H1 = styled.h1`
`;

export default App;
