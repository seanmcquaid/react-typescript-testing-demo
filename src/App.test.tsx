import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';
import App from './App';

describe("<App/>", () => {
  afterEach(cleanup);
  test("Renders correctly", () => {
    const app = render(<App />);

    expect(app).toMatchSnapshot();
  });

  test("Text Input displays correctly when changed", () => {
    const { getByTestId } = render(<App />);
    
    fireEvent.change(getByTestId("textInput"), { target: { value: "Test Input Value" } });
    
    expect(getByTestId("textInput")).toHaveValue("Test Input Value");
  });

  test("Button on click will add an item", () => {
    const { getByTestId, getByText } = render(<App />);
    
    fireEvent.change(getByTestId("textInput"), { target: { value: "Test Input Value" } });

    fireEvent.click(getByText("Click Me!"));

    expect(getByTestId("itemNumber0")).toBeInTheDocument();
    expect(getByTestId("textInput")).toHaveValue("");
  });
});
