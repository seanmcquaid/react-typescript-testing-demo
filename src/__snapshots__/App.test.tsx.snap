// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> Renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <h1
          class="sc-AxiKw cydMkg"
        >
          Testing Tutorial
        </h1>
        <input
          class="sc-AxjAm fDEmJb"
          data-testid="textInput"
          value=""
        />
        <button
          class="sc-AxirZ dqVvws"
        >
          Click Me!
        </button>
        <ul />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <h1
        class="sc-AxiKw cydMkg"
      >
        Testing Tutorial
      </h1>
      <input
        class="sc-AxjAm fDEmJb"
        data-testid="textInput"
        value=""
      />
      <button
        class="sc-AxirZ dqVvws"
      >
        Click Me!
      </button>
      <ul />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
